ckb-contracts/lib/eth-bridge-lockscript-lib/src/adapter/chain.rs

```diff
@@ -0,0 +1,62 @@
+use super::{Adapter, BridgeCellDataTuple, ComplexData};
+use ckb_std::ckb_constants::Source;
+use ckb_std::error::SysError;
+use ckb_std::high_level::{
+    load_cell_data, load_cell_lock_hash, load_script_hash, load_tx_hash, QueryIter,
+};
+
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+
+pub struct ChainAdapter {}
+
+impl Adapter for ChainAdapter {
+    fn load_tx_hash(&self) -> Result<[u8; 32], SysError> {
+        load_tx_hash()
+    }
+
+    fn load_input_output_data(&self) -> Result<BridgeCellDataTuple, SysError> {
+        fn load_data_from_input() -> Option<Vec<u8>> {
+            let data_list =
+                QueryIter::new(load_cell_data, Source::GroupInput).collect::<Vec<Vec<u8>>>();
+            match data_list.len() {
+                0 => None,
```

Comment:
It should never be None.
Because lockscript only executes when it appears in inputs.